### PR TITLE
improve-hookfs

### PR DESCRIPTION
**Pull request by qiffang**
Function
- Add rename hook
In prometheus tsdb, it needs this scene

- Use go module
Use go-fuse 1.0, but interface of nodefs.File are changed
```
//implements nodefs,GetLk
func (h *hookFile) GetLk(owner uint64, lk *fuse.FileLock, flags uint32, out *fuse.FileLock) (code fuse.Status) {
	return h.file.GetLk(owner, lk, flags, out)
}

//implements nodefs.SetLk
func (h *hookFile) SetLk(owner uint64, lk *fuse.FileLock, flags uint32) (code fuse.Status) {
	return h.file.SetLk(owner, lk, flags)
}

//implements nodefs,SetLkw
func (h *hookFile) SetLkw(owner uint64, lk *fuse.FileLock, flags uint32) (code fuse.Status) {
	return h.file.SetLkw(owner, lk, flags)
}
```

- Support test case
Actually, we do not need to unmount mountpoint by the manual way.
We can unmount it by this way in test case
```
syscall.Unmount(mountpoint, -1) //of course, it also call fusermount command
```

### CODE REVIEW

File: cmd/ex01/main.go [hunk 1]
```diff
@@ -3,22 +3,20 @@ package main
 import (
 	"flag"
 	"fmt"
+	"github.com/qiffang/hookfs/pkg/example"
```
Comment (AkihiroSuda) [inline cmd/ex01/main.go]: osrg/hooks ?

**Comment by AkihiroSuda** [comment]
thanks, could you also sign the commit with `git commit -s`?

File: cmd/ex01/main.go (hunk 1 above)
Comment (qiffang) [reply cmd/ex01/main.go]: Updated

**Comment by qiffang** [comment]
> thanks, could you also sign the commit with `git commit -s`?

Done and thanks

**Pull request merged**